### PR TITLE
feat: add `--channel-priority={strict|disabled}`

### PR DESCRIPTION
**Pull request by tl-hbk**
I needed a way to change the channel priority due to having a similar situation as the op of this issue https://github.com/prefix-dev/rattler-build/issues/343

I'm new to rust so there's probably a better way of adding it as a cli arg without the wrapper but I couldn't figure it out

### CODE REVIEW

File: src/opt.rs [hunk 1]
```diff
@@ -189,6 +190,29 @@ pub struct CommonOpts {
...
+}
+
+impl FromStr for ChannelPriorityWrapper {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "strict" => Ok(ChannelPriorityWrapper { value: ChannelPriority::Strict }),
+            "disabled" => Ok(ChannelPriorityWrapper { value: ChannelPriority::Disabled }),
+            _ => Err(format!("Unknown channel priority: {}", s))
```
Comment (wolfv) [inline src/opt.rs:213]: For a user it would be nice to list the available options for a perfectly helpful error message :)

**Comment by wolfv** [comment]
Looks good to me! Left one small comment.

File: src/opt.rs (hunk 1 above)
Comment (tl-hbk) [reply src/opt.rs]: I've updated the error message

**Review by wolfv — APPROVED** [review]

**Comment by tl-hbk** [comment]
@wolfv it looks like some workflows need to be approved before they can run.  Could you approve those?

**Comment by wolfv** [comment]
For some weird reason the button to run the tests does not show up. Can you push an empty commit?

**Comment by tl-hbk** [comment]
> For some weird reason the button to run the tests does not show up. Can you push an empty commit?

done

**Comment by wolfv** [comment]
OK, finally CI is running! :)

**Comment by happysalada** [comment]
Hi was there a specific test with channel priority disabled ?
Im still getting "is excluded because due to strict channel priority not using this"

I am specifying several channels as arguments from the cli, could that be the problem ?

**Comment by tl-hbk** [comment]
I didn't add a test to the repo but after the new version was released I tested it also using several channels as cli args and it worked fine for me

```
rattler-build build -r <recipe> -c <channel 1> -c <channel 2> --channel-priority disabled
```

**Pull request merged**